src/plugins/planout.ts

```diff
@@ -0,0 +1,364 @@
+        this.namespaces[namespaceName] = ns.clone();
+      });
+    }
+  }
+
+  addNamespace(namespace: string, numSegments: number = 1000, primaryUnit: string = 'uid', compiled?: PlanoutCode) {
+    if (!this.namespaces[namespace]) {
+      const ns = new BaseNamespace(namespace);
+      this.namespaces[namespace] = ns;
+      ns.setName(namespace);
+      ns.numSegments = numSegments;
+      ns.setPrimaryUnit(primaryUnit);
+      ns.setDefaultExperimentConfig({
+        namespace_name: namespace,
+        name: '_default',
+        compiled: compiled || { op: 'seq', seq: [] },
+        num_segments: 0,
+      });
+    }
+    return this.namespaces[namespace];
+  }
+
+  add(experimentConfig: PlanoutExperimentConfig) {
+    const namespace = experimentConfig.namespace_name;
+    if (!this.namespaces[namespace]) {
+      this.addNamespace(namespace);
+    }
+    this.namespaces[namespace].addExperimentConfig(experimentConfig);
+  }
+
+  remove(namespace: string, name: string) {
+    if (this.namespaces[namespace]) {
+      this.namespaces[namespace].removeExperiment(name);
+    }
+  }
+
+  useConfig(config: PlanoutConfig) {
+    this.resetNamespaces();
+    config.namespaces.forEach((nsConfig) => {
+      this.addNamespace(
+        nsConfig.name,
+        nsConfig.num_segments,
+        nsConfig.primary_unit,
+        nsConfig.compiled);
+    });
+
+    config.experiments.forEach((nsOp) => {
+      const experiment = nsOp.experiment;
+      if (nsOp.op === 'add') {
+        this.add(experiment);
+      } else if (nsOp.op === 'remove') {
+        this.remove(experiment.namespace_name, experiment.name);
+      } else {
+        console.error(`Unknown operation ${nsOp.op}.`);
+      }
+    });
+
+    if (options.localOverrides) {
+      Object.keys(options.localOverrides).forEach((namespace: string) => {
+        const nsOverrides = options.localOverrides![namespace];
+        const ns = this.namespaces[namespace] || this.addNamespace(namespace);
+        nsOverrides.forEach((override) => {
+          ns.setOverride('default', override.variable, override.value);
+        });
+      });
+    }
+  }
+
+  async fetchConfig() {
+    try {
+      const response = await this.axios.get<PlanoutConfig>(`/cms/planout/config/${options.appName}`);
+      const config = await options.getConfig();
+      if (!isEqual(response.data, config)) {
+        console.info('Planout: Experiment config updated. Re-initializing.');
+        options.setConfig(response.data);
+        this.useConfig(response.data);
+      } else {
+        console.info('Planout: Experiment config unchanged.');
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+}
+
+export const planout = new PlanoutPlugin();
+
+export default (Vue: typeof _Vue, options: PlanoutPluginOptions): void => {
+  planout.setOptions(options);
+
+  Object.defineProperties(Vue.prototype, {
+    $planout: {
+      get() {
+        planout.fetchConfig();
```

Comment:
I'm a bit confused by this line. Does this say that every time we do `this.$planout.<whatever>`, it'll fetch the planout config?